keps/sig-node/5328-node-capabilities/README.md

```diff
@@ -0,0 +1,891 @@
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes a new **Node Capabilities** framework for Kubernetes nodes to declaratively inform the control plane about the static capabilities and features they support, such as specific Kubelet, kernel, and runtime configurations. This information is designed to help control plane components like kube-scheduler and/or admission controllers to make better decisions.
+
+For **scheduling**, the `kube-scheduler` would utilize these advertised Node Capabilities by matching a workload's requirements (inferred from the pod spec) against them. This ensures pods are only placed on nodes that definitively possess the necessary features to run them successfully.
+
+For **API request validation**, admission controllers would leverage these Node Capabilities. This allows them to validate incoming API requests and prevent operations if a node lacks the required static features.
+
+The intent of this Node Capabilities framework is to prevent both the scheduling of pods onto incompatible nodes and the execution of unsupported operations or configurations on specific nodes. This, in turn, aims to streamline cluster operations by reducing the reliance on manual configurations like taints, tolerations, and complex node labeling schemes for managing node-specific features.
+
+## Motivation
+
+This proposal includes "Node Capabilities" in Kubernetes to ensure pods are scheduled and can operate reliably on nodes while reducing the operational burden. It provides a standardized way for Kubelet to advertise specific node features and configurations, decreasing reliance on manual Taints or Node Labels.
+
+Node Capabilities aims to prevent pods from being scheduled on incompatible nodes - those missing necessary features because of version skew between control plane and the node or unsupported runtime/kernel configurations ([slack discussion](https://kubernetes.slack.com/archives/C5P3FE08M/p1741867194258139)). Making the scheduler aware of specific node capabilities will enable more reliable pod placement and ensure that incompatibilities are proactively identified as scheduling failures.
+
+### Goals
+
+1. Define a standard mechanism for nodes to expose Kubelet, Runtime, and Kernel configurations that are pertinent to workload scheduling and/or improve API Request validation. 
+2. Enhance the kube-scheduler to understand pod requirements and match them against Node capabilities and place pods on compatible nodes. 
+3. Enable API validation mechanisms to utilize Node Capabilities to verify support for requested operations (e.g., in-place resize, ephemeral containers) before they are processed.
```

Comment:
does it mean we will expose capabilities for ephemeral containers? Or that we will "enable" possibility to add it? I am not clear here what is the scope of work